src/App.tsx

```diff
@@ -1,17 +1,99 @@
+import { useState } from "react";
 import Header from "./Components/Header";
-import InputTodo from "./Components/InputTodo";
-import TodoList from "./Components/TodoList";
+import TodoInput from "./Components/TodoInput";
+import TodoItem from "./Components/TodoItem";
+import CreatePortal from "./Components/CreatePortal";
+import { useAppDispatch, useAppSelector } from "./hooks";
+import { selectItem } from "./Features/textItem/item-slice";
+import { IItem } from "./Features/textItem/item-slice";
+
+import {
+  addItem,
+  deleteItem,
+  editItem,
+  completedItem,
+} from "./Features/textItem/item-slice";
 
 const App: React.FC = () => {
+  const [showEdit, setShowEdit] = useState<boolean>(false);
+  const [edit, setEdit] = useState<string>("");
+  const [editId, setEditId] = useState<number>();
+  const [saveEdit, setSaveEdit] = useState<string>("");
+
+  const dispatch = useAppDispatch();
+
+  const addTodosHandler = (initialState: IItem) => {
+    setEditId(initialState.id);
+    dispatch(addItem(initialState));
+  };
+
+  const todo = useAppSelector(selectItem);
+
+  const editElementHandler = (initialState: IItem) => {
+    setShowEdit((previousState: boolean) => !previousState);
+    setEditId(initialState.id);
+
+    setEdit(initialState.todo);
+    setSaveEdit(initialState.todo);
+  };
+
+  const deleteElementHandler = (initialState: IItem) => {
+    dispatch(deleteItem(initialState));
+  };
+
+  // МОДАЛЬНОЕ ОКНО
+  //__________________________________________________________
+  const editValueHandler = (textInput: string) => {
+    setEdit(textInput);
+  };
+
+  const cancelShowHandler = () => {
+    setShowEdit((previousState: boolean) => false);
```

Comment:
as you don't use previousState, there is no need to declare it, can be simplified to usetShowEdit(() => false) or just usetShowEdit(false)